src/vs/workbench/contrib/terminal/browser/terminalInstance.ts

```diff
@@ -1631,6 +1633,10 @@ export class TerminalInstance extends Disposable implements ITerminalInstance {
 			}
 		}
 
+		if (shellIntegrationAttempted) {
+			this._telemetryService.publicLog2<{ classification: 'SystemMetaData'; purpose: 'FeatureInsight' }>('terminal/shellIntegrationProcessExit');
+		}
```

Comment:
This doesn't seem useful to me as all shell integration attempted terminals will exit eventually, it's possible this code would not run though (eg. crash).

Reply:
it's useful because it tells us whether the process crashed upon creation when trying to launch with shell integration

https://github.com/microsoft/vscode/blob/549c4077ecb6bd89299d8547441fd67a8e9390ff/src/vs/workbench/contrib/terminal/browser/terminalInstance.ts#L1548-L1550

Reply:
Oh I was misunderstanding, can we change `shellIntegrationAttempted` to `failedShellIntegrationInjection` or something more descriptive? I thought it just meant that the process exited after shell integration args were injected (whether that failed or not).